### PR TITLE
Deprecated decorator implemented

### PR DESCRIPTION
**Pull request by enchev**

### CODE REVIEW

File: declarations.d.ts [hunk 1]
```diff
@@ -16,7 +16,23 @@ declare var console: Console;
...
+    set?: (value: T) => void;
+}
+
+declare type ClassDecorator = <TFunction extends Function>(target: TFunction) => TFunction | void;
+declare type PropertyDecorator = (target: Object, propertyKey: string | symbol) => void;
+declare type MethodDecorator = <T>(target: Object, propertyKey: string | symbol, descriptor: TypedPropertyDescriptor<T>) => TypedPropertyDescriptor<T> | void;
+declare type ParameterDecorator = (target: Object, propertyKey: string | symbol, parameterIndex: number) => void;
+
 // Global functions
+declare function Deprecated(target: Object, key?: string | symbol, value?: any): void;
```
Comment (ligaz) [inline declarations.d.ts]: [Here](https://github.com/Microsoft/TypeScript/blob/2cb0dfd99dc2896958b75e44303d8a7a32e5dc33/tests/cases/conformance/decorators/class/accessor/decoratorOnClassAccessor1.ts#L3) is the correct signature.

**Comment by dtopuzov** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

**Comment by nsndeck** [comment]
:+1:

**Comment by hamorphis** [comment]
:+1:

**Pull request merged**